packages/firestore/lite/src/api/reference.ts

```diff
@@ -1233,3 +1247,14 @@ export function newUserDataReader(
     serializer
   );
 }
+
+export function newExpUserDataWriter<T>(
+  firestore: FirebaseFirestore,
+  converter: UntypedFirestoreDataConverter<T> | null
+): UserDataWriter {
+  return new UserDataWriter(
```

Comment:
A more traditional arrangement would be to have `UserDataWriter` be an abstract class and have different subclasses supply the conversion routines by implementing methods that are missing. Then you wouldn't need a `newExpUserDataWriter` function, you'd just invoke the constructor of the subtype instead.

Reply:
I updated the PR as suggested (which would match what we discussed last week). I was a bit lazy in the first iteration, as the factory functions were already implemented.